### PR TITLE
feat: Add a step to validate the filename

### PR DESCRIPTION
**Pull request by benjaminParisel**
* check if the filename follow the convention (kebab-case)
* suggest a new filename to resolve the issue.

Covers https://github.com/bonitasoft/bonita-documentation-site/issues/589

### CODE REVIEW

**Comment by benjaminParisel** [comment]
Tested with https://github.com/bonitasoft/bonita-documentation-site/pull/791 PR.
Publish a comment on PR with this display when the page filename step is on error.

image

File: packages/doc-contribs/pr-antora-content-guidelines-checker/src/steps/PageFilenameStep.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    }
+
+    let commentBody = `## 🥙 ${this.name} \n`;
+    commentBody += `${this.description}\n`;
+    this.stepResult.results.forEach((actionResult) => {
+      commentBody += `- [ ] Rename **${actionResult.file}** to **${actionResult.details}**\n`;
+    });
+    commentBody += `
+> [!warning]
+> Additionally, don't forget to add the \`:page-aliases:\` attribute after renaming the **existing files** to avoid broken links or references. `;
```
Comment (tbouffard) [inline packages/doc-contribs/pr-antora-content-guidelines-checker/src/steps/PageFilenameStep.ts:39]: **suggestion:** Here the warning applies to file that previously existed prior the PR creation.
If the file is a new file created in the PR, there is no need to add an alias (and it shouldn't be added).
Let's talk in person to find a more convenient message.

File: packages/doc-contribs/pr-antora-content-guidelines-checker/__tests__/steps/PageFilenameStep.test.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    const pageFilenameStep = new PageFilenameStep(files, extensionsToCheck);
+
+    const result = await pageFilenameStep.validate(mockOctokit);
+
+    expect(result).toEqual({
+      status: Status.ERROR,
+      results: [
+        { file: "testAreNotOk.adoc", details: "test-are-not-ok.adoc" },
+        { file: "name_are_bad.adoc", details: "name-are-bad.adoc" },
+        { file: "otherBad_filename.adoc", details: "other-bad-filename.adoc" },
```
Comment (tbouffard) [inline packages/doc-contribs/pr-antora-content-guidelines-checker/__tests__/steps/PageFilenameStep.test.ts:49]: **praise:** 👍🏿 for testing the kebab-case transformation here with several use cases!

**Review by tbouffard — APPROVED** [review]
LGTM, great work! 🎉 

I haven't approved the PR yet, because I'd first like to talk about the current situation:
- the warning message about eventual alias creation
- if we do a manual rename operations in all repositories prior merging/releasing this new feature. This is what was planned in https://github.com/bonitasoft/bonita-documentation-site/issues/589

**Comment by tbouffard** [comment]
> LGTM, great work! 🎉
> 
> I haven't approved the PR yet, because I'd first like to talk about the current situation:
> 
>     * the warning message about eventual alias creation
> 
>     * if we do a manual rename operations in all repositories prior merging/releasing this new feature. This is what was planned in [Tool: enforce file name to use kebab-case in documentation content bonita-documentation-site#589](https://github.com/bonitasoft/bonita-documentation-site/issues/589)

Answering to my own questions:
* warning message: I am going to propose an alternative
* the actions provides a way to skip the step. It is used with a reusable workflow in the documentation content repositories, so by adding a new input to the reusable workflow, it will be possible to disable the step.

**Pull request merged**